core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfo.scala

```diff
@@ -17,26 +40,245 @@ sealed abstract class UTXOSpendingInfo {
   /** The funding transaction's txid and the index of the output in the transaction we are spending */
   def outPoint: TransactionOutPoint
 
+  def amount: CurrencyUnit
+
+  def scriptPubKey: ScriptPubKey
+
   /** the actual output itself we are spending */
-  def output: TransactionOutput
+  def output: TransactionOutput = {
+    TransactionOutput(value = amount, scriptPubKey = scriptPubKey)
+  }
 
   /** the signers needed to spend from the output above */
   def signers: Seq[Sign]
 
-  /** a redeemScript, if required, to spend the output above */
+  def hashType: HashType
+
   def redeemScriptOpt: Option[ScriptPubKey]
 
-  /** the scriptWitness, if required, to spend the output above */
   def scriptWitnessOpt: Option[ScriptWitness]
+}
 
-  def hashType: HashType
+sealed trait BitcoinUTXOSpendingInfo extends UTXOSpendingInfo
+
+object BitcoinUTXOSpendingInfo {
+
+  // TODO: Get rid of this and force all callers to directly call a subclass apply method
```

Comment:
Do we want to do this on this PR or a subsequent one? It seems fine to me to use as a shim for now, especially with all these other PRs we have open.

Reply:
I was thinking subsequent

Reply:
too keep things small(er)